src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/fahimi2018/PropDisjunctive.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature.fahimi2018;
+
+import org.chocosolver.solver.constraints.nary.cumulative.literature.CumulativeFilter;
+import gnu.trove.list.array.TIntArrayList;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * Disjunctive constraint filtering algorithms described in the following paper :
+ * Fahimi, H., Ouellet, Y., Quimper, C.-G.: Linear-Time Filtering Algorithms for the Disjunctive Constraint and a Quadratic Filtering Algorithm for the Cumulative Not-First Not-Last. Constraints 23(3), pages 272–293 (2018). https://doi.org/10.1007/s10601-018-9282-9
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class PropDisjunctive extends CumulativeFilter {
+    private Integer[] Ip;
```

Comment:
Could this be turned into `int[]` instead ?

Reply:
Not because of this line (that appears two times in the class) : 
Arrays.sort(Ip, Comparator.comparingInt(i -> tasks[i].getDuration().getLB()));

i needs to be an Object